tests/TestCase/ORM/TableTest.php

```diff
@@ -2394,36 +2410,94 @@ public function testSaveUpdatePrimaryKeyNotModified()
             ->method('execute')
             ->will($this->returnValue($statement));
 
-        $query->expects($this->once())->method('set')
-            ->with(['username' => 'baggins'])
+        $query->expects($this->once())
+            ->method('update')
+            ->will($this->returnValue($query));
+        
+        $query->expects($this->once())
+            ->method('set')
+            ->with($mergedData)
+            ->will($this->returnValue($query));
+        
+        $query->expects($this->once())
+            ->method('where')
+            ->with(['id' => 2])
             ->will($this->returnValue($query));
 
-        $entity = new \Cake\ORM\Entity([
-            'id' => 2,
-            'username' => 'baggins'
-        ], ['markNew' => false]);
-        $this->assertSame($entity, $table->save($entity));
+        $changedEntity = $table->patchEntity($entity, $newData);
+        $resultingEntity = $table->save($changedEntity);
+
+        $this->assertNotSame($originalData, $resultingEntity->toArray());
+        $this->assertSame($mergedData, $resultingEntity->toArray());
     }
 
     /**
-     * Tests that passing only the primary key to save will not execute any queries
-     * but still return success
+     * Tests that passing only the primary key to save will execute a query
      *
      * @group save
      * @return void
      */
-    public function testUpdateNoChange()
-    {
+    public function testUpdateKeyChange()
+    {
+        $originalData = ['id' => 3, 'username' => 'Kamille',
+            'password' => '$2a$10$u05j8FjsvLBNdfhBhc21LOuVMpzpabVXQ9OpC2wO3pSO0q6t7HHMO',
+            'created' => '2012-06-10 01:22:23', 'updated' => '2015-05-21 15:43:00'];
+        $newData = ['id' => 2];
+        $mergedData = ['id' => 2, 'username' => 'Kamille',
+            'password' => '$2a$10$u05j8FjsvLBNdfhBhc21LOuVMpzpabVXQ9OpC2wO3pSO0q6t7HHMO',
+            'created' => '2012-06-10 01:22:23', 'updated' => '2015-05-21 15:43:00'];
+        
+        $entity = $this->getMock(
+            '\Cake\ORM\Entity',
+            null,
+            [$originalData, ['markNew' => false]]
+        );
+        
+        $this->assertSame($originalData, $entity->toArray());
```

Comment:
Well, originally the old test was doing so, would you prefer that I modify existing data in the Fixture?